### PR TITLE
fix: fix manifest reader extensions, ensure that all builtins go through npe2

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
Fixes the extensions declared by the built-in reader plugin.  Needed for #4564.

This turned into a bigger PR because of the number of places where we were accidentally(?) relying on the fact that our builtins plugins were being provided by both npe2 and npe1.  This ensures we're not using npe1 internally anymore by removing the decorators on our plugins.

closes https://github.com/napari/napari/pull/4580

included in #4669

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by DragaDoncila** [comment]
@tlambert03 sorry for the double work but I think this PR is equivalent to #4580, except that #4580 also reverts the workaround we merged before I realized the issue was with our manifest 🤦‍♀️

**Comment by tlambert03** [comment]
ok, I see why this wasn't done before 😂 
just had to fix a number of issues with "builtins" vs "napari".  It looks like the "absence" of all of these extensions in the manifest has led to an unfortunate co-existence of our builtins  plugin in both the new and old plugin managers... but there's no reason "builtins" needs to exist anywhere at all anymore (we depend on npe2 and can assume it's installed).  I updated this PR to fix all those tests.

The only place the word "builtins" really needs to be supported is in the public API, and I've just added a line that converts "builtins" to "napari" with a warning in all public functions.


@DragaDoncila, would appreciate your eyes on this one

File: napari/plugins/_tests/test_save_layers.py
Comment (DragaDoncila) [inline napari/plugins/_tests/test_save_layers.py]: Should this line not also be deleted?

File: napari/plugins/_tests/test_utils.py
Comment (DragaDoncila) [inline napari/plugins/_tests/test_utils.py:48]: So glad to see this nonsense disappear 🙏 😭

File: napari/plugins/_tests/test_utils.py
Comment (DragaDoncila) [inline napari/plugins/_tests/test_utils.py:57]: I think here we would be better off reverting #4575 (as we do in #4580). This PR supersedes #4580

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
Thanks for taking on this most annoying of PRs @tlambert03 😆 . I left just a couple of comments and have a few more here.

1. I'd love it if we could change the display name for `napari` to either `builtins` or `napari builtins`. I think it's weird when opening files and choosing plugins in the GUI to see `napari` as a choice, and I think `builtins` is clearer. I think @jni was of this opinion as well
2. I think you should revert #4575 in this PR and close #4580 after, as this supersedes it. 
3. I think you should get rid of the lines in `plugins/utils/get_potential_readers` and `plugins/utils/get_all_readers` that check if `builtins` and `napari` are both present.
4. Because we convert `builtins` to `napari` and warn in `viewer.open` this shouldn't affect saved preferences right?

Very much looking forward to that shim...

**Comment by tlambert03** [comment]
ok, getting closer.
@draga, there are still three tests failing in test_viewer, but I'm not entirely sure how to fix them (i.e. what the expected behavior was).  They seem to be explicitly testing builtins _being_ an npe1 plugin?  No hurry, but 👀 appreciated

**Comment by tlambert03** [comment]
side note: I see we have a number of plugin related tests in `napari/components` now... it'd be nice if we could keep all plugin (even builtin) related tests in napari/plugins, maybe in a latter PR?

File: napari/plugins/_tests/test_utils.py
Comment (tlambert03) [reply napari/plugins/_tests/test_utils.py:57]: did it, I guess after you reviewed?

**Comment by tlambert03** [comment]
ok, I skipped two tests that are particularly tricky with the changes here (since they essentially required that builtin readers only come from the npe1 plugin manager).  While they are somewhat important tests, they're also new tests, and could be modified in the near future.  I'll leave it to @DragaDoncila  to decide whether we keep them skipped or fix them here

File: napari/_qt/widgets/_tests/test_qt_extension2reader.py
Comment (DragaDoncila) [inline napari/_qt/widgets/_tests/test_qt_extension2reader.py:123]: @tlambert03 this one at least we can fix - I *think* the full test should be:

```python
def test_filtering_readers(
    extension2reader_widget, qtbot, tmp_reader, mock_npe2_pm
):
    tmp_reader(mock_npe2_pm, 'npy-reader', filename_patterns=['*.npy'])
    tmp_reader(mock_npe2_pm, 'tif-reader', filename_patterns=['*.tif'])

    widget = extension2reader_widget()

    assert widget._new_reader_dropdown.count() == 2
    widget._filter_compatible_readers('*.npy')
    assert widget._new_reader_dropdown.count() == 1
    all_dropdown_items = [
        widget._new_reader_dropdown.itemText(i)
        for i in range(widget._new_reader_dropdown.count())
    ]
    assert all_dropdown_items == ['npy-reader']
```

File: napari/components/viewer_model.py
Comment (DragaDoncila) [inline napari/components/viewer_model.py]: Is this leftover from debugging?

File: napari/_qt/widgets/_tests/test_qt_extension2reader.py
Comment (tlambert03) [reply napari/_qt/widgets/_tests/test_qt_extension2reader.py:123]: no biggie.  I'm gonna skip it here.  i refactored the fixtures in https://github.com/napari/napari/pull/4669

File: napari/components/viewer_model.py
Comment (tlambert03) [reply napari/components/viewer_model.py]: lol.  eek

**Review by DragaDoncila — APPROVED** [review]
@tlambert03 this all looks good from my end. I think the skipped tests can be skipped for now, and the way we should fix them is by making use of `tmp_reader` with a reader function - I did initially play with that before using builtins and had some trouble, but we can explore that separately.

Thank you for this PR, it's definitely not a fun one 🙏

**Comment by tlambert03** [comment]
@napari/core-devs, I'd like to get this one in sooner than later (since #4564, #4669 and #4673 depend on it).  Will merge in 24 hours unless anyone objects and/or merges it sooner :)

**Pull request merged**
